packages/react/README.md

```diff
@@ -1,26 +1,125 @@
 
-See the [Example Package](https://www.npmjs.com/package/@jsonforms/examples) on how to integrate JSONForms with your application.
+See the official [documentation](https://jsonforms.io/) and the JSON Forms React [seed repository](https://github.com/eclipsesource/jsonforms-react-seed) for examples on how to integrate JSON Forms with your application.
+
+Check <https://www.npmjs.com/search?q=%40jsonforms> for all published JSON Forms packages.
+
+### Usage
+
+Use the `JsonForms` component to render a form for your data.
+
+Mandatory props:
+
+* `data: any` - the data to show
+* `renderers: JsonFormsRendererRegistryEntry[]` - the React renderer set to use
+
+Optional props:
+
+* `schema: JsonSchema` - the data schema for the given data. Will be generated when not given.
+* `uischema: UISchemaElement` - the UI schema for the given data schema. Will be generated when not given.
+* `cells: JsonFormsCellRendererRegistryEntry[]` - the React cell renderer set to use
+* `config: any` - form-wide options. May contain default ui schema options.
+* `readonly: boolean` - whether all controls shall be readonly.
+* `uischemas: JsonFormsUiSchemaEntry[]` - registry for dynamic ui schema dispatching
+* `validationMode: 'ValidateAndShow' | 'ValidateAndHide' | 'NoValidation'` - the validation mode for the form
+* `ajv: AJV` - custom Ajv instance for the form
+* `refParserOptions: RefParserOptions` - configuration for ref resolving
+* `onChange` - callback which is called on each data change, containing the updated data and the validation result.
+
+Example:
+
+```ts
+import React, { useState } from 'react';
+import {
+  materialRenderers,
+  materialCells
+} from '@jsonforms/material-renderers';
+import { JsonForms } from '@jsonforms/react';
+
+const schema = {
+  type: 'object',
+  properties: {
+    name: {
+      type: 'string',
+      minLength: 1
+    },
+    done: {
+      type: 'boolean'
+    },
+    due_date: {
+      type: 'string',
+      format: 'date'
+    },
+    recurrence: {
+      type: 'string',
+      enum: ['Never', 'Daily', 'Weekly', 'Monthly']
+    }
+  },
+  required: ['name', 'due_date']
+};
+const uischema = {
+  type: 'VerticalLayout',
+  elements: [
+    {
+      type: 'Control',
+      label: false,
+      scope: '#/properties/done'
+    },
+    {
+      type: 'Control',
+      scope: '#/properties/name'
+    },
+    {
+      type: 'HorizontalLayout',
+      elements: [
+        {
+          type: 'Control',
+          scope: '#/properties/due_date'
+        },
+        {
+          type: 'Control',
+          scope: '#/properties/recurrence'
+        }
+      ]
+    }
+  ]
+};
+const data = {};
+function App() {
+  return (
+    <JsonForms
+      schema={schema}
+      uischema={uischema}
+      data={data}
+      renderers={materialRenderers}
+      cells={materialCells}
+    />
+  );
+}
```

Comment:
enhance the example to place data into a `useState` and update it via `onChange`.